src/dev/i18n/config.ts

```diff
@@ -21,45 +21,47 @@ import { resolve } from 'path';
 
 // @ts-ignore
 import { normalizePath, readFileAsync } from '.';
-// @ts-ignore
-import rootConfig from '../../../.i18nrc.json';
 
 export interface I18nConfig {
   paths: Record<string, string>;
   exclude: string[];
   translations: string[];
+  prefix?: string;
 }
 
-/**
- * Merges root .i18nrc.json config with any other additional configs (e.g. from
- * third-party plugins).
- * @param configPaths List of config paths.
- */
-export async function mergeConfigs(configPaths: string | string[] = []) {
-  const mergedConfig: I18nConfig = { exclude: [], translations: [], ...rootConfig };
+export async function checkConfigNamespacePrefix(configPath: string) {
+  const { prefix, paths } = JSON.parse(await readFileAsync(resolve(configPath)));
+  for (const [namespace] of Object.entries(paths)) {
+    if (prefix && prefix !== namespace.split('.')[0]) {
+      throw new Error(`namespace ${namespace} must be prefixed with ${prefix} in ${configPath}`);
+    }
+  }
+}
 
-  for (const configPath of Array.isArray(configPaths) ? configPaths : [configPaths]) {
-    const additionalConfig: I18nConfig = {
-      paths: {},
-      exclude: [],
-      translations: [],
-      ...JSON.parse(await readFileAsync(resolve(configPath))),
-    };
+export async function mergeConfig(
+  configPath: string,
+  config: I18nConfig = { exclude: [], translations: [], paths: {} }
+) {
+  const additionalConfig: I18nConfig = {
+    paths: {},
+    exclude: [],
+    translations: [],
+    ...JSON.parse(await readFileAsync(resolve(configPath))),
+  };
 
-    for (const [namespace, path] of Object.entries(additionalConfig.paths)) {
-      mergedConfig.paths[namespace] = normalizePath(resolve(configPath, '..', path));
-    }
+  for (const [namespace, path] of Object.entries(additionalConfig.paths)) {
+    config.paths[namespace] = normalizePath(resolve(configPath, '..', path));
+  }
 
-    for (const exclude of additionalConfig.exclude) {
-      mergedConfig.exclude.push(normalizePath(resolve(configPath, '..', exclude)));
-    }
+  for (const exclude of additionalConfig.exclude) {
+    config.exclude.push(normalizePath(resolve(configPath, '..', exclude)));
+  }
 
-    for (const translations of additionalConfig.translations) {
-      mergedConfig.translations.push(normalizePath(resolve(configPath, '..', translations)));
-    }
+  for (const translations of additionalConfig.translations) {
+    config.translations.push(normalizePath(resolve(configPath, '..', translations)));
   }
 
-  return mergedConfig;
+  return config;
```

Comment:
This `mergeConfig` mutates the incoming `config` object. Do you foresee any bugs arising from this, or would it be better to pull items from `config` into `additionalConfig` and return that, so as to avoid the mutation?

Reply:
The name `mergeConfig` indicates mutation. Eventually I'll have to mutate the config file passed in the context or override it so it can be passed to other `ListR` tasks.

Reply:
> The name `mergeConfig` indicates mutation

I don't agree, and from what I can tell the return value is taking the place of the previous `config` variable (even though that's technically a noop) so I would be quite surprised to find out that `mergeConfig()` mutated it's second argument.

This implies mutation to me:
```js
context.config = ...;
await assignConfigFromPath(context.config, configPath);
```

This doesn't:
```js
context.config = await mergeConfig(configPath, context.config);
```